File: xm/insertion.c
#include<stdio.h>

void insertion_sort(int a[], int length);

int main()
{
	int a[]={8,4,9,5,7,6,3,2,},i;
	insertion_sort(a, 8);
	for (i = 0;i < 8;i++ )
	{
		printf("a[%d] = %d\n",i,a[i]);
	}
	
	return 0;
}

void insertion_sort(int a[], int length)
{
	int i, j, key;
	
	for(i = 1;i < length;i++)
	{
		key = a[i];
		j = i-1;
		while(j >= 0 && a[j] > key)
		{
			a[j+1] = a[j];
			j = j-1;
		}
		a[j+1] = key;
	}
}

File: Practical Python/insertion.py
def insertion_sort(list):
    for i in range (1,len(list)):
        for j in range(i-1,0,-1):
            if (list[j]>list[j+1]):
                list[j],list[j+1]=list[j+1],list[j]
            else:
                break
    return(list)


n=int(input("Enter the no of element to sort:\t"))
myList=[]
print("Enter element one by one: \n")
for x in range(n):
    k=int(input(f"Element {x}:\t"))
    myList.append(k)
print(f"Unordered List: {myList}")
print(f"After using Bouble sort on the list: \n{insertion_sort(myList)}")